src/edtr-io/plugins/image.tsx

```diff
@@ -85,44 +93,102 @@ export function createUploadImageHandler(getCsrfToken: () => string) {
   }
 }
 
-export function createReadFile(getCsrfToken: () => string) {
+export function createReadFile() {
   return function readFile(file: File): Promise<LoadedFile> {
     return new Promise((resolve, reject) => {
-      const reader = new FileReader()
-
-      reader.onload = function (e: ProgressEvent) {
-        if (!e.target) return
-        const formData = new FormData()
-        formData.append('attachment[file]', file)
-        formData.append('type', 'file')
-        formData.append('csrf', getCsrfToken())
-
-        fetch('/attachment/upload', {
-          method: 'POST',
-          body: formData,
-        })
-          .then((response) => response.json())
-          .then((data: { success: boolean; files: { location: string }[] }) => {
-            if (!data['success']) reject()
-            resolve({
-              file,
-              dataUrl: data.files[0].location,
-            })
-          })
-          .catch(() => {
-            reject()
+      const authenticationPayload = parseAuthCookie()
+
+      if (!authenticationPayload?.token) return
+
+      const gqlFetch = createAuthAwareGraphqlFetch({
+        current: authenticationPayload,
+      })
+      const args = JSON.stringify({
+        query: uploadUrlQuery,
+        variables: { mediaType: MediaType.ImagePng },
+      })
+      void gqlFetch(args).then((data: MediaUploadQuery) => {
+        console.log('resulting url:')
+        console.log(data.media.newUpload.uploadUrl)
+        const reader = new FileReader()
+
+        reader.onload = function (e: ProgressEvent) {
+          if (!e.target) return
+          const formData = new FormData()
+          formData.append('attachment[file]', file)
+          formData.append('type', 'file')
+
+          fetch(data.media.newUpload.uploadUrl, {
+            method: 'PUT',
+            headers: { 'Content-Type': 'image/jpeg' }, // TODO: fixed only for testing
+            body: formData,
```

Comment:
```suggestion
            body: file,
```

Directly add the file as content instead of using form data (Google doesn't support it)